backend/assimilator.py

```diff
@@ -55,7 +55,41 @@ def assimilate(data_type, data_path, data_idName, geo_path, geo_idName, out_path
                 boundary["properties"] = data_toAssimilate[properties[geo_idName]]
 
             data_assimilated = boundaries
-    # print(boundary["properties"].keys())
+
+    # If input is points
+    elif data_type == "points":
+        point_features = []
+
+        #Read in csv as list of dictionaries so each row is a dictionary with {headers : values}
+        with open(data_path, newline="", encoding="utf8") as f:
+            reader = csv.DictReader(f)
+            
+            try:
+                for row in reader:
+                    properties = []
+
+                    for key in row:
+                        try:
+                            properties.append((key, float(row[key].replace(",", "",))))
+                        except:
+                            properties.append((key, row[key]))
+                    
+                    #For each point (row in csv), detect and extract all headers (dictionary keys), store values, and add to list of dictionaries
+                    properties = {key: value for key, value in properties}
+
+                    #Form geojson format for points, dump properties as dictionaries into properties
+                    point_features.append({
+                        "type": "Feature",
+                        "geometry": {"type": "Point", "coordinates": [float(row['Lng']), float(row['Lat'])]},
+                        "properties": properties,
+                    })
+            except:
+                print("WARNING (Assimilator): Unable to assimilate row: ", row)
+            
+            print("Message (Assimilator): Found {} points for {} ".format(len(point_features),data_path))
+            
+            #Output list of features (with properties) in geojson format; convert to list for json serialisation by using sorted()
+            data_assimilated = {"type": "FeatureCollection", "features": point_features}
```

Comment:
This is another _bad smell_ : in a complex pipeline like this one it is advisable to rely on a _well-defined_ data object (that _can_ builds on Dictionary and their flexibility) but having a clearly defined data (exchange) interface is better for maintainability and modularity.

Reply:
Okay thanks valerio for the information about 💩s,

Reply:
This is not 💩 - this is SE encapsulation and proper OOP design